packages/block-editor/src/components/inner-blocks/index.js

```diff
@@ -58,12 +59,12 @@ class InnerBlocks extends Component {
 	}
 
 	componentDidUpdate( prevProps ) {
-		const { template, block } = this.props;
+		const { template, block, templateLock } = this.props;
 		const { innerBlocks } = block;
 
 		this.updateNestedSettings();
-		// only synchronize innerBlocks with template if innerBlocks are empty or a locking all exists
-		if ( innerBlocks.length === 0 || this.getTemplateLock() === 'all' ) {
```

Comment:
Is `getTemplateLock` useful now? Should we remove inheritance entirely?

I have to admit, I'm having trouble understanding how locking in nested context work in relationship with the global template, and the template provided as prop... If I'm having trouble understanding it, others will. I'm wondering if we should have a document that clarifies how it works?

Reply:
What sort of documentation would we need to unblock the PR? An example section in the handbook?

Reply:
To clarify, I don't think documentation should block the PR. I was just having hard time understanding how this work, especially thinking of all contexts. (CPT template, prop template, CPT lock, prop lock...)

Reply:
Hi @youknowriad,
> Is getTemplateLock useful now?

We only had one usage of it. I guess we can remove and inline its logic. The change was applied.

>Should we remove inheritance entirely?

Nice question, I think the inheritance mechanism is useful. E.g.: without it, if I use group with some paragraphs inside or a cover in a CPT with locking all, inside the group or cover the user will always be able to do anything and locking has no effect.
If the user has a locking all and wants in a specific cover block to allow the user to do anything that's also possible nowadays. The developer can create a container block that sets locking to false and in the template set the cover to be inside that block.
So I think inheritance together with child/container blocks provides a powerful mechanism and removing inheritance is removing some options.


Yes, I agree that the logic is complex and hard to understand. I will create a separate PR that tries to document how these mechanisms work so we don't block this PR.

Reply:
So With these changes, if we want to lock the synchronization we need to set it up explicitly?

```es6
<div className="wp-block-cover__inner-container">
	<InnerBlocks
		template={ INNER_BLOCKS_TEMPLATE }
		templateLock="all"
	/>
</div>
```

Just FYI, I've been working on previewing different blocks using the `<BlockPreview />` component, getting into issues specifically with `core/cover` and `core/image-text` blocks.
This PR fixes the problem.